dynamicdns.go

```diff
@@ -252,7 +252,7 @@ func (a App) checkIPAndUpdateDNS() {
 			)
 		}
 		for _, rec := range records {
-			name := joinDomainZone(rec.Name, zone)
+			name := libdns.AbsoluteName(rec.Name, zone)
```

Comment:
Note that this isn't strictly the same because `AbsoluteName` returns a trailing `.` in the result. This might have side effects :thinking:

We need to make sure everything that uses the result doesn't expect to not have the trailing dot. Did you trace through the code to make sure it's only used inside the program and isn't used as input elsewhere? (I don't have time to dig into it right now but I'll follow up later)

Reply:
Yeah this appears to only be used as key in maps of IP addresses. We don't rely on the actual values, only that they uniquely identify a DNS record.